staging/src/k8s.io/kubectl/pkg/cmd/expose/expose.go

```diff
@@ -378,3 +385,186 @@ func (o *ExposeServiceOptions) RunExpose(cmd *cobra.Command, args []string) erro
 	}
 	return nil
 }
+
+func (o *ExposeServiceOptions) createService() (*corev1.Service, error) {
+	if len(o.Selector) == 0 {
```

Comment:
As mentioned before move this to `Validate` method which you don't have, yet.

Reply:
See comment above